### ISSUE TITLE
lines-between-class-members should not apply to class fields

### ISSUE THREAD

**Issue by p-bakker** [eslint/eslint#11153]
**What rule do you want to change?**
lines-between-class-members

**Does this change cause the rule to produce more or fewer warnings?**
Fewer

**How will the change be implemented? (New option, new default behavior, etc.)?**
new default behavior

**Please provide some example code that this change will affect:**

```js
class x {
  #a 
  #b
  #c

  get a() {
    return this#a
  }

  someMethod() {
    ...
  }
}
```

**What does the rule currently do for this code?**
Enforce empty lines between the declared (private) class fields

**What will the rule do after it's changed?**
Not enforce new lines between class field declaration

**Are you willing to submit a pull request to implement this change?**
I would if I could figure out how rules work...

**Comment by g-plane** [comment]
Class fields proposal is in Stage 3, which ESLint doesn't support.

**Comment by ljharb** [comment]
Either way, I'd expect this rule to definitely be able to apply to class fields - but for fields and methods to be separately configurable.

<sub>also note the syntax is `this.#a`/`obj.#a` etc, and the declarations can have initializers (`#a = 3;`, eg)</sub>

**Comment by platinumazure** [comment]
Labeling this as "experimental" for now since this syntax is Stage 3. If this is promoted to Stage 4, this can be reopened and we can evaluate (and hopefully accept) the proposal. Thanks!
